src/snowflake/cli/api/commands/flags.py

```diff
@@ -350,8 +350,15 @@ def _password_callback(value: str):
     rich_help_panel=_CONNECTION_SECTION,
 )
 
+
+def _diag_log_path_callback(path: str):
+    if path is not None:
+        return path
+    return tempfile.gettempdir()
+
+
 DiagLogPathOption: Path = typer.Option(
-    tempfile.gettempdir(),
+    None,
```

Comment:
Let's add a comment explaining why we do so

Reply:
Done!